src/write/loc.rs

```diff
@@ -0,0 +1,496 @@
+                        let address = convert_address(addr)?;
+                        Location::BaseAddress { address }
+                    }
+                    read::RawLocListEntry::BaseAddressx { addr } => {
+                        have_base_address = true;
+                        let address = convert_address(context.dwarf.address(context.unit, addr)?)?;
+                        Location::BaseAddress { address }
+                    }
+                    read::RawLocListEntry::StartxEndx { begin, end, data } => {
+                        let begin = convert_address(context.dwarf.address(context.unit, begin)?)?;
+                        let end = convert_address(context.dwarf.address(context.unit, end)?)?;
+                        let data = Expression(data.0.to_slice()?.into());
+                        Location::StartEnd { begin, end, data }
+                    }
+                    read::RawLocListEntry::StartxLength {
+                        begin,
+                        length,
+                        data,
+                    } => {
+                        let begin = convert_address(context.dwarf.address(context.unit, begin)?)?;
+                        let data = Expression(data.0.to_slice()?.into());
+                        Location::StartLength {
+                            begin,
+                            length,
+                            data,
+                        }
+                    }
+                    read::RawLocListEntry::OffsetPair { begin, end, data } => {
+                        let data = Expression(data.0.to_slice()?.into());
+                        Location::OffsetPair { begin, end, data }
+                    }
+                    read::RawLocListEntry::StartEnd { begin, end, data } => {
+                        let begin = convert_address(begin)?;
+                        let end = convert_address(end)?;
+                        let data = Expression(data.0.to_slice()?.into());
+                        Location::StartEnd { begin, end, data }
+                    }
+                    read::RawLocListEntry::StartLength {
+                        begin,
+                        length,
+                        data,
+                    } => {
+                        let begin = convert_address(begin)?;
+                        let data = Expression(data.0.to_slice()?.into());
+                        Location::StartLength {
+                            begin,
+                            length,
+                            data,
+                        }
+                    }
+                    read::RawLocListEntry::DefaultLocation { data } => {
+                        let data = Expression(data.0.to_slice()?.into());
+                        Location::DefaultLocation { data }
+                    }
+                };
+                // In some cases, existing data may contain begin==end, filtering
+                // these out.
+                match loc {
```

Comment:
The range.rs has no such logic. The fixtures/self/debug_loc contains entries with such invalid data.

Reply:
Looks good, maybe we should add this to range.rs too.